test/library/input.spec.js

```diff
@@ -56,6 +56,33 @@ describe('NumberFormat as input', () => {
     expect(wrapper.find('input').instance().value).toEqual('$2,456,981');
   });
 
+  it('should load the default value when initial value is null', () => {
+    const wrapper = mount(<NumberFormat value={null} defaultValue={89} />);
+    expect(wrapper.state().value).toEqual('89');
+  });
+
+  it('should load the prevous valid value if the state is changed to null', () => {
+    class WrapperComponent extends React.Component {
+      constructor() {
+        super ();
+        this.state = {
+          testState: 90,
+        };
+      }
+      render() {
+        return (<NumberFormat value={this.state.testState} />)
+      }
+    }
+
+    const wrapper = mount(<WrapperComponent />);
+    const input = wrapper.find('input');
+    const domInput = input.instance();
+
+    expect(domInput.value).toEqual('90');
+    wrapper.setState({testState: null});
```

Comment:
This spec could be simplified you just mount `<NumberFormat value={this.state.testState} />` and then change the props through setProps.

Also FYI. Try avoiding interacting with states in specs. You shouldn't depend much on implementation detail.